### PR TITLE
feat(typescript): converting task files to typescript

### PR DESCRIPTION
**Pull request by cnishina**

### CODE REVIEW

File: lib/taskLogger.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  /**
+   * Log output such that metadata are appended.
+   * Calling log(data) will not flush to console until you call flush()
+   *
+   * @constructor
+   * @param {object} task Task that is being reported.
+   * @param {number} pid PID of process running the task.
+   */
+  constructor(task: any, pid: number) {
```
Comment (alexeagle) [inline lib/taskLogger.ts]: you can declare the property here as well, with `private task: any` - now you don't need to declare it above, or set the value in the body

File: lib/taskLogger.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    this.task = task;
+    this.pid = pid;
+    this.buffer = '';
```
Comment (alexeagle) [inline lib/taskLogger.ts]: assign the default value at the declaration site - it is called immediately following the constructor

File: lib/taskRunner.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+   * './runner.js') or from a new process (via './runnerCli.js').
+   *
+   * @constructor
+   * @param {string} configFile Path of test configuration.
+   * @param {object} additionalConfig Additional configuration.
+   * @param {object} task Task to run.
+   * @param {boolean} runInFork Whether to run test in a forked process.
+   * @constructor
+   */
+  constructor(configFile, additionalConfig, task, runInFork) {
```
Comment (alexeagle) [inline lib/taskRunner.ts]: same as above, can be shorter

File: lib/taskRunner.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+      // stdout pipe
+      childProcess.stdout.on('data', (data) => { taskLogger.log(data); });
+
+      // stderr pipe
+      childProcess.stderr.on('data', (data) => { taskLogger.log(data); });
+
+      childProcess
+          .on('message',
+              (m) => {
```
Comment (alexeagle) [inline lib/taskRunner.ts]: hmm, any particular reasoning that you left noImplicitAny as false?

**Comment by alexeagle** [comment]
LG for typescript

File: lib/taskLogger.ts (hunk 1 above)
Comment (cnishina) [inline lib/taskLogger.ts]: Done.

File: lib/taskLogger.ts (hunk 2 above)
Comment (cnishina) [inline lib/taskLogger.ts]: Done.

File: lib/taskRunner.ts (hunk 3 above)
Comment (cnishina) [inline lib/taskRunner.ts]: Done.

File: lib/taskRunner.ts (hunk 4 above)
Comment (cnishina) [inline lib/taskRunner.ts]: Changed it back to true.

**Comment by juliemr** [comment]
Let's change the commit name to `chore(typescript)` - with the reasoning that it's not a feature that someone reading the changelog would care about.

File: lib/taskScheduler.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
+import ConfigParser, {Config} from './configParser';
```
Comment (juliemr) [inline lib/taskScheduler.ts]: Should `ConfigParser` be in the`{}`?

Comment (juliemr) [inline lib/taskScheduler.ts]: Or is this the default thing working as intended?

File: tsconfig.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
     "sourceMap": false,
     "declaration": true,
     "removeComments": false,
-    "noImplicitAny": false,
```
Comment (juliemr) [inline tsconfig.json]: Why this change? I feel like we might want to be more lenient about this unless we have to be strict.

**Comment by juliemr** [comment]
Looks good other than a couple questions.

File: lib/taskScheduler.ts (hunk 5 above)
Comment (cnishina) [inline lib/taskScheduler.ts]: The class ConfigParser is labeled as the default and does not require `{}`.

File: tsconfig.json (hunk 6 above)
Comment (cnishina) [inline tsconfig.json]: `noImplicitAny` is helpful to ensure types in the method signature. If we need to be lenient, we could use `any` as the type.

Comment (juliemr) [inline tsconfig.json]: Sorry, I misread the boolean. LGTM!

**Comment by cnishina** [comment]
Reference feature request: #1203

**Comment by cnishina** [comment]
Closed with commit 608104b
